torch/csrc/jit/tensor_conversions.h

```diff
@@ -61,7 +61,7 @@ struct tensor_as_impl<at::IntList> {
   at::IntList operator()(at::Tensor&& t) {
     if (t.type().scalarType() != at::ScalarType::Long)
       throw tensor_conversion_error("Expected a LongTensor");
-    if (t.dim() != 1)
+    if (t.dim() > 1)
```

Comment:
I don't think this is actually fixing the code correctly. View, reshape, and repeat allow varargs `view(1,2,3)` rather than `view([1,2,3])`. Allow `dim() == 0` here only handles the case of `view(3)`. It is also surprising to allow a 0-dim tensor to become an int list. It would be like allow `1` to become `std::vector<int>`.  This PR fixes these bugs by handling varargs correct (https://github.com/pytorch/pytorch/pull/7444) but it is a bit messy and I have not had time to clean it up.

Reply:
Ok couple things:

1. The issue you describe is covered in https://github.com/pytorch/pytorch/issues/8516, should we merge this into that task?
2. Also, Wanchao is taking on that task, is your PR a good starting point for him to push through the fix?